src/services/identity/OtpService.ts

```diff
@@ -19,8 +19,10 @@ class OtpService {
     return { otp, hashedOtp }
   }
 
-  verifyOtp = async (otp: string, hashedOtp: string): Promise<boolean> =>
-    bcrypt.compare(otp, hashedOtp)
+  verifyOtp = async (otp: string, hashedOtp: string): Promise<boolean> => {
+    if (!otp || !hashedOtp) return false
+    return bcrypt.compare(otp, hashedOtp)
```

Comment:
the documentation states `A promise to be either resolved with the comparison result salt or rejected with an Error`

should we add a try catch, and return `false` if there was any error thrown?

Reply:
The return type of the function is a Promise<boolean> and is the same as bcrypt compare

```
export declare function compare(data: string | Buffer, encrypted: string): Promise<boolean>;
```

So for the upstream users of this method, either way it will resolve to an boolean